hail/python/hail/expr/expressions/base_expression.py

```diff
@@ -754,6 +754,116 @@ def _to_relational_preserving_rows_and_cols(self):
             return x.order_by(*x.key).select(name)
         return x
 
+    def export(self, filename, delimiter='\t', header=True):
```

Comment:
I think we generally use `path` as the variable for this, can we use that out of consistency? Also add typechecks.